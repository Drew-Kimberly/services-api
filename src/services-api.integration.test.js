const axios = require('axios').default;

const baseUrl = 'http://localhost:3100/api';

describe('GET /services', () => {
    it('should return all service definitions', async () => {
        const response = await axios.get(`${baseUrl}/services`);
        expect(response.status).toEqual(200);
        expect(response.data).toMatchSnapshot();
    });

    it('should sort associated version arrays by version ID desc', async () => {
        const response = await axios.get(`${baseUrl}/services`);
        expect(response.status).toEqual(200);
        const services = response.data.data;
        services.forEach(svc => {
            let currentVersionId = Number.MAX_SAFE_INTEGER;
            svc.versions.forEach(version => {
                expect(version.id).toBeLessThan(currentVersionId);
                currentVersionId = version.id;
            });
        });
    });

    describe('search', () => {
        it('should support wildcard search by name', async () => {
            const searchTerm = 'baz';
            const response = await axios.get(`${baseUrl}/services?search=${searchTerm}`);

            expect(response.status).toEqual(200);
            const services = response.data.data;
            expect(services.length).toEqual(2);
            services.forEach(svc => {
                expect(svc.name.toLowerCase()).toEqual(expect.stringContaining(searchTerm));
            });
        });

        it('should support wildcard search by description', async () => {
            const searchTerm = 'nana';
            const response = await axios.get(`${baseUrl}/services?search=${searchTerm}`);
            expect(response.status).toEqual(200);
            const services = response.data.data;
            expect(services.length).toEqual(1);
            expect(services[0].description.toLowerCase()).toEqual(expect.stringContaining(searchTerm));
        });

        it('should return an empty result set when the search does not match', async () => {
            const searchTerm = 'does-not-exist';
            const response = await axios.get(`${baseUrl}/services?search=${searchTerm}`);
            expect(response.status).toEqual(200);
            const services = response.data.data;
            expect(services.length).toEqual(0);
        });
    });

    describe('sort', () => {
        it('should default to sort by service ID desc', async () => {
            const response = await axios.get(`${baseUrl}/services`);
            expect(response.status).toEqual(200);

            const services = response.data.data;
            let currentServiceId = Number.MAX_SAFE_INTEGER;
            services.forEach(svc => {
                expect(svc.id).toBeLessThan(currentServiceId);
                currentServiceId = svc.id;
            });
        });

        it('should change direction based on the sortDir query param', async () => {
            const response = await axios.get(`${baseUrl}/services?sortDir=asc`);
            expect(response.status).toEqual(200);

            const services = response.data.data;
            let currentServiceId = Number.MIN_SAFE_INTEGER;
            services.forEach(svc => {
                expect(svc.id).toBeGreaterThan(currentServiceId);
                currentServiceId = svc.id;
            });
        });

        it('should change direction based on the sortDir query param', async () => {
            const response = await axios.get(`${baseUrl}/services?sortDir=asc`);
            expect(response.status).toEqual(200);

            const services = response.data.data;
            let currentServiceId = Number.MIN_SAFE_INTEGER;
            services.forEach(svc => {
                expect(svc.id).toBeGreaterThan(currentServiceId);
                currentServiceId = svc.id;
            });
        });

        it('should change the field sorted by based on the sortBy query param', async () => {
            const response = await axios.get(`${baseUrl}/services?sortDir=asc&sortBy=name`);
            expect(response.status).toEqual(200);

            const services = response.data.data;
            let currentServiceName = 'a';
            services.forEach(svc => {
                expect(svc.name.localeCompare(currentServiceName)).toBeGreaterThanOrEqual(0);
                currentServiceName = svc.name;
            });
        });

        it('should not sort by unsupported fields or directions', async () => {
            const response = await axios.get(`${baseUrl}/services?sortDir=bad&sortBy=versions`);
            expect(response.status).toEqual(200);

            const services = response.data.data;
            let currentServiceId = Number.MAX_SAFE_INTEGER;
            services.forEach(svc => {
                expect(svc.id).toBeLessThan(currentServiceId);
                currentServiceId = svc.id;
            });
        });

        it('should include sort metadata in the response', async () => {
            const sortField = 'updatedAt';
            const sortDir = 'desc';
            const response = await axios.get(`${baseUrl}/services?sortDir=${sortDir}&sortBy=${sortField}`);
            expect(response.status).toEqual(200);
            expect(response.data.meta).toEqual(
                expect.objectContaining({
                    sortField,
                    sortDir
                })
            );
        });
    });

    describe('pagination', () => {
        it('should support limiting the returned results', async () => {
            const limit = 5;
            const response = await axios.get(`${baseUrl}/services?limit=${limit}`);
            expect(response.status).toEqual(200);
            expect(response.data.data.length).toEqual(limit);
        });

        it('should support paging through limited results', async () => {
            const limit = 5;
            const page1Response = await axios.get(`${baseUrl}/services?limit=${limit}&page=1`);
            expect(page1Response.status).toEqual(200);
            const page2Response = await axios.get(`${baseUrl}/services?limit=${limit}&page=2`);
            expect(page2Response.status).toEqual(200);

            expect(page1Response.data.data.length).toEqual(page2Response.data.data.length);
            expect(page1Response.data.data).not.toEqual(page2Response.data.data);
        });

        it('should return respect search and sort while paging', async () => {
            const limit = 2;
            const sortField = 'name';
            const search = 'ba';

            const page1Response = await axios.get(
                `${baseUrl}/services?limit=${limit}&page=1&sortBy=${sortField}&search=${search}`
            );
            expect(page1Response.status).toEqual(200);
            const page2Response = await axios.get(
                `${baseUrl}/services?limit=${limit}&page=2&sortBy=${sortField}&search=${search}`
            );
            expect(page2Response.status).toEqual(200);

            let currentServiceName = 'z';
            page1Response.data.data.forEach(svc => {
                expect(`${svc.name}--${svc.description}`.toLowerCase()).toEqual(expect.stringContaining(search));
                expect(svc.name.localeCompare(currentServiceName)).toBeLessThanOrEqual(0);
                currentServiceName = svc.name;
            });

            page2Response.data.data.forEach(svc => {
                expect(`${svc.name}--${svc.description}`.toLowerCase()).toEqual(expect.stringContaining(search));
                expect(svc.name.localeCompare(currentServiceName)).toBeLessThanOrEqual(0);
                currentServiceName = svc.name;
            });
        });

        it('should return remaining results when paginated to the final page', async () => {
            const totalRecords = 15; // Heavy assumption :'(
            const limit = 10;
            const lastPage = Math.ceil(totalRecords / limit);
            const expectedLastPageCount = totalRecords % limit;

            const response = await axios.get(`${baseUrl}/services?limit=${limit}&page=${lastPage}`);
            expect(response.status).toEqual(200);
            expect(response.data.data.length).toEqual(expectedLastPageCount);
        });

        it('should include pagination metadata in the response', async () => {
            const totalRecords = 15; // Heavy assumption :'(
            const limit = 5;
            const currentPage = 2;
            const response = await axios.get(`${baseUrl}/services?limit=${limit}&page=${currentPage}`);
            expect(response.status).toEqual(200);
            expect(response.data.meta).toEqual(
                expect.objectContaining({
                    returnedCount: limit,
                    totalCount: totalRecords,
                    pageCount: Math.ceil(totalRecords / limit),
                    page: currentPage
                })
            );
        });

        it('should reflect adjusted record count in metadata when search is included', async () => {
            const searchTerm = 'bar';
            const limit = 2;
            const page = 2;
            const response = await axios.get(`${baseUrl}/services?limit=${limit}&page=${page}&search=${searchTerm}`);
            expect(response.status).toEqual(200);
            expect(response.data.meta).toEqual(
                expect.objectContaining({
                    returnedCount: 1,
                    totalCount: 3,
                    pageCount: 2,
                    page
                })
            );
        });
    });
});

describe('GET /services/:serviceId', () => {
    it('should return the request service definition', async () => {
        const serviceId = 1;
        const response = await axios.get(`${baseUrl}/services/${serviceId}`);
        expect(response.status).toEqual(200);
        expect(response.data).toMatchSnapshot();
    });

    it('should sort the associated version array by version ID desc', async () => {
        const serviceId = 5;
        const response = await axios.get(`${baseUrl}/services/${serviceId}`);
        expect(response.status).toEqual(200);

        const service = response.data.data;
        let currentVersionId = Number.MAX_SAFE_INTEGER;
        service.versions.forEach(version => {
            expect(version.id).toBeLessThan(currentVersionId);
            currentVersionId = version.id;
        });
    });

    it('should return a 404 given a valid but unknown service ID', async () => {
        const serviceId = 999;
        try {
            await axios.get(`${baseUrl}/services/${serviceId}`);
        } catch (e) {
            expect(e.response.status).toEqual(404);
            return;
        }

        throw new Error('Should not reach!');
    });

    it('should return a 404 given an invalid service ID', async () => {
        const serviceId = 'should-not-be-valid';
        try {
            await axios.get(`${baseUrl}/services/${serviceId}`);
        } catch (e) {
            expect(e.response.status).toEqual(404);
            return;
        }

        throw new Error('Should not reach!');
    });
});

describe('GET /services/:serviceId/versions', () => {
    it('should return all versions for a given service', async () => {
        const serviceId = 1;
        const response = await axios.get(`${baseUrl}/services/${serviceId}/versions`);
        expect(response.status).toEqual(200);
        expect(response.data).toMatchSnapshot();
    });

    it('should return a 404 given an unknown Service ID', async () => {
        const serviceId = 321;
        try {
            await axios.get(`${baseUrl}/services/${serviceId}/versions`);
        } catch (e) {
            expect(e.response.status).toEqual(404);
            return;
        }

        throw new Error('Should not reach!');
    });

    it('should sort by Version ID DESC by default', async () => {
        const serviceId = 1;
        const response = await axios.get(`${baseUrl}/services/${serviceId}/versions`);
        expect(response.status).toEqual(200);

        const versions = response.data.data;
        let currentVersionId = Number.MAX_SAFE_INTEGER;
        versions.forEach(version => {
            expect(version.id).toBeLessThan(currentVersionId);
            currentVersionId = version.id;
        });
    });

    it('should return response metadata for sort/pagination', async () => {
        const serviceId = 1;
        const limit = 1;
        const sortField = 'name';
        const response = await axios.get(
            `${baseUrl}/services/${serviceId}/versions?limit=${limit}&sortBy=${sortField}`
        );
        expect(response.status).toEqual(200);
        expect(response.data.meta).toEqual(
            expect.objectContaining({
                returnedCount: limit,
                totalCount: 2,
                pageCount: 2,
                page: 1,
                sortDir: 'desc',
                sortField
            })
        );
    });
});

describe('POST /services', () => {
    const baseService = {
        name: 'Integration Test Service POST',
        description: 'Service created by a POST request',
        versions: []
    };

    it('should create a new service', async () => {
        const response = await axios.post(`${baseUrl}/services`, baseService);
        expect(response.status).toEqual(201);
        expect(response.data).toMatchSnapshot();

        const fetchResponse = await axios.get(`${baseUrl}/services/${response.data.data.id}`);
        expect(fetchResponse.status).toEqual(200);
        expect(response.data.data).toEqual(expect.objectContaining(baseService));
    });

    it('should throw a 400 when provided a service that already exists', async () => {
        const service = { ...baseService, id: 25 };

        try {
            await axios.post(`${baseUrl}/services`, service);
        } catch (e) {
            expect(e.response.status).toEqual(400);
            return;
        }

        throw new Error('Should not reach!');
    });

    it('should return an error response when given an empty name field', async () => {
        const service = { ...baseService, name: '' };
        await expect(() => axios.post(`${baseUrl}/services`, service)).rejects.toThrow();
    });

    it('should return an error response when a name field is too long', async () => {
        const service = { ...baseService, name: new Array(150).fill('a').join('') };
        await expect(() => axios.post(`${baseUrl}/services`, service)).rejects.toThrow();
    });

    it('should return an error response when description field is too long', async () => {
        const service = { ...baseService, name: new Array(600).fill('a').join('') };
        await expect(() => axios.post(`${baseUrl}/services`, service)).rejects.toThrow();
    });
});

describe('PUT /services/:serviceId', () => {
    const baseService = {
        name: 'Integration Test Service PUT',
        description: 'Service upserted by a PUT request',
        versions: []
    };

    it('should create a new service if that service does not exist', async () => {
        const serviceId = 17;
        const response = await axios.put(`${baseUrl}/services/${serviceId}`, baseService);
        expect(response.status).toEqual(201);

        const fetchResponse = await axios.get(`${baseUrl}/services/${serviceId}`);
        expect(fetchResponse.status).toEqual(200);
        expect(fetchResponse.data.data).toEqual(expect.objectContaining(baseService));
    });

    it('should update an existing service idempotently', async () => {
        const serviceId = 17;
        const updatedName = baseService.name + ' updated';
        const updatedService = { ...baseService, name: updatedName };

        const response1 = await axios.put(`${baseUrl}/services/${serviceId}`, updatedService);
        expect(response1.status).toEqual(200);

        const response2 = await axios.put(`${baseUrl}/services/${serviceId}`, updatedService);
        expect(response2.status).toEqual(200);

        const fetchResponse = await axios.get(`${baseUrl}/services/${serviceId}`);
        expect(fetchResponse.status).toEqual(200);
        expect(fetchResponse.data.data).toEqual(expect.objectContaining(updatedService));
    });

    it('should return a 404 given an invalid service ID', async () => {
        const serviceId = 'should-not-be-valid';
        const service = {
            name: 'test-service',
            description: 'My service created by tests',
            versions: []
        };

        try {
            await axios.put(`${baseUrl}/services/${serviceId}`, service);
        } catch (e) {
            expect(e.response.status).toEqual(404);
            return;
        }

        throw new Error('Should not reach!');
    });

    it('should return an error response when given an empty name field', async () => {
        const serviceId = 5;
        const service = { ...baseService, name: '' };
        await expect(() => axios.put(`${baseUrl}/services/${serviceId}`, service)).rejects.toThrow();
    });

    it('should return an error response when a name field is too long', async () => {
        const serviceId = 5;
        const service = { ...baseService, name: new Array(150).fill('a').join('') };
        await expect(() => axios.put(`${baseUrl}/services/${serviceId}`, service)).rejects.toThrow();
    });

    it('should return an error response when description field is too long', async () => {
        const serviceId = 5;
        const service = { ...baseService, name: new Array(600).fill('a').join('') };
        await expect(() => axios.put(`${baseUrl}/services/${serviceId}`, service)).rejects.toThrow();
    });
});

describe('DELETE /services/:serviceId', () => {
    it('should delete the specified service', async () => {
        const serviceId = 16;
        const response = await axios.delete(`${baseUrl}/services/${serviceId}`);
        expect(response.status).toEqual(204);

        try {
            await axios.get(`${baseUrl}/services/${serviceId}`);
        } catch (e) {
            expect(e.response.status).toEqual(404);
            return;
        }

        throw new Error('Should not reach!');
    });

    it('should be idempotent', async () => {
        const serviceId = 17;
        const response1 = await axios.delete(`${baseUrl}/services/${serviceId}`);
        expect(response1.status).toEqual(204);

        const response2 = await axios.delete(`${baseUrl}/services/${serviceId}`);
        expect(response2.status).toEqual(204);
    });

    it('should return a 404 given an invalid service ID', async () => {
        const serviceId = 'should-not-be-valid';
        try {
            await axios.delete(`${baseUrl}/services/${serviceId}`);
        } catch (e) {
            expect(e.response.status).toEqual(404);
            return;
        }

        throw new Error('Should not reach!');
    });
});
