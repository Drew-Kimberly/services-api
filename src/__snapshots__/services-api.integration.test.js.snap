// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /services should return all service definitions 1`] = `
Object {
  "data": Array [
    {
  "id": 15,
  "name": "Reporting",
  "description": "pears",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 30,
      "name": "v3",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 15
    },
    {
      "id": 29,
      "name": "v2",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 15
    }
  ]
},
    {
  "id": 14,
  "name": "Users",
  "description": "strawberries",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 28,
      "name": "v1",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 14
    },
    {
      "id": 27,
      "name": "v3",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 14
    }
  ]
},
    {
  "id": 13,
  "name": "Service123",
  "description": "oranges",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 26,
      "name": "v2",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 13
    },
    {
      "id": 25,
      "name": "v1",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 13
    }
  ]
},
    {
  "id": 12,
  "name": "My Service",
  "description": "bananas",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 24,
      "name": "v3",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 12
    },
    {
      "id": 23,
      "name": "v2",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 12
    }
  ]
},
    {
  "id": 11,
  "name": "Reporting",
  "description": "apples",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 22,
      "name": "v1",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 11
    },
    {
      "id": 21,
      "name": "v3",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 11
    }
  ]
},
    {
  "id": 10,
  "name": "FooBarBaz",
  "description": "8e18444e204954e0fe9ef049278269a1",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 20,
      "name": "v2",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 10
    },
    {
      "id": 19,
      "name": "v1",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 10
    }
  ]
},
    {
  "id": 9,
  "name": "BarBaz",
  "description": "",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 18,
      "name": "v3",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 9
    },
    {
      "id": 17,
      "name": "v2",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 9
    }
  ]
},
    {
  "id": 8,
  "name": "FooBar",
  "description": "c302fb83c9512c6e10d0c40608f91eee",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 16,
      "name": "v1",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 8
    },
    {
      "id": 15,
      "name": "v3",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 8
    }
  ]
},
    {
  "id": 7,
  "name": "Priority Services",
  "description": "3ef79689dff252ba68d6f15df4056de0",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 14,
      "name": "v2",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 7
    },
    {
      "id": 13,
      "name": "v1",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 7
    }
  ]
},
    {
  "id": 6,
  "name": "Notifications",
  "description": "8409ae41c8693b0a60c819a8cd9a5089",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 12,
      "name": "v3",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 6
    },
    {
      "id": 11,
      "name": "v2",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 6
    }
  ]
},
    {
  "id": 5,
  "name": "FX Rates International",
  "description": "",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 10,
      "name": "v1",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 5
    },
    {
      "id": 9,
      "name": "v3",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 5
    }
  ]
},
    {
  "id": 4,
  "name": "FX Rates International",
  "description": "6bb2e62a4012839467747cee994da475",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 8,
      "name": "v2",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 4
    },
    {
      "id": 7,
      "name": "v1",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 4
    }
  ]
},
    {
  "id": 3,
  "name": "Contact Us",
  "description": "f3b930ad7edf872bea01cb15240fe785",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 6,
      "name": "v3",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 3
    },
    {
      "id": 5,
      "name": "v2",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 3
    }
  ]
},
    {
  "id": 2,
  "name": "Collect Monday",
  "description": "afd73373999a6b19c5a96db0bcec82aa",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 4,
      "name": "v1",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 2
    },
    {
      "id": 3,
      "name": "v3",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 2
    }
  ]
},
    {
  "id": 1,
  "name": "Locate Us",
  "description": "93cc8046e4d1db11aa909d3f14573487",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 2,
      "name": "v2",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 1
    },
    {
      "id": 1,
      "name": "v1",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 1
    }
  ]
},
  ],
  "meta": Object {
    "page": 1,
    "pageCount": 1,
    "returnedCount": 15,
    "sortDir": "desc",
    "sortField": "id",
    "totalCount": 15,
  },
}
`;

exports[`GET /services/:serviceId should return the request service definition 1`] = `
Object {
  "data": {
  "id": 1,
  "name": "Locate Us",
  "description": "93cc8046e4d1db11aa909d3f14573487",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "versions": [
    {
      "id": 2,
      "name": "v2",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 1
    },
    {
      "id": 1,
      "name": "v1",
      "createdAt": "mockCreatedAtTimestamp",
      "updatedAt": "mockUpdatedAtTimestamp",
      "serviceId": 1
    }
  ]
},
  "meta": Object {},
}
`;

exports[`GET /services/:serviceId/versions should return all versions for a given service 1`] = `
Object {
  "data": Array [
    {
  "id": 2,
  "name": "v2",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "serviceId": 1
},
    {
  "id": 1,
  "name": "v1",
  "createdAt": "mockCreatedAtTimestamp",
  "updatedAt": "mockUpdatedAtTimestamp",
  "serviceId": 1
},
  ],
  "meta": Object {
    "page": 1,
    "pageCount": 1,
    "returnedCount": 2,
    "sortDir": "desc",
    "sortField": "id",
    "totalCount": 2,
  },
}
`;

exports[`POST /services should create a new service 1`] = `
Object {
  "data": {
  "id": 16,
  "name": "Integration Test Service POST",
  "description": "Service created by a POST request",
  "versions": [],
  "updatedAt": "mockUpdatedAtTimestamp",
  "createdAt": "mockCreatedAtTimestamp"
},
  "meta": Object {},
}
`;
